cmd/avalanche-network-runner/control/control.go

```diff
@@ -100,10 +101,16 @@ func newStartCommand() *cobra.Command {
 		"[optional] JSON string of map that maps from VM to its genesis file path",
 	)
 	cmd.PersistentFlags().StringVar(
-		&nodeConfig,
-		"node-config",
+		&defaultNodeConfig,
+		"default-node-config",
 		"",
-		"node config as string",
+		"[optional] default node config as JSON string, applied to all nodes",
+	)
+	cmd.PersistentFlags().StringVar(
+		&customNodeConfigs,
+		"custom-node-configs",
+		"",
+		"[optional] custom node configs as JSON string of map, for each node individually. Common entries verride `default-node-config`, but can be combined. Invalidates `number-of-nodes` (provide all node configs if uses).",
```

Comment:
In principle I don't see the need to invalidate number-of-nodes. You can just override some nodes. Or am I missing something?
Regarding the help, a suggestion:
"[optional] custom configs for each node. with precedence over `default-node-config`. invalidates `number-of-nodes` (you must provide configs for all nodes)."